diff --git a/chrome/android/BUILD.gn b/chrome/android/BUILD.gn
index 02f17d2b4e93d27964fe6663c4feddccc9634619..21cc5020894f8c03f685ca11d2ff72e55679d69c 100644
--- a/chrome/android/BUILD.gn
+++ b/chrome/android/BUILD.gn
@@ -38,7 +38,7 @@ if (enable_vr) {
   import("//chrome/android/features/vr/vr_module_tmpl.gni")
 }
 
-manifest_package = "org.chromium.chrome"
+declare_args() { manifest_package = "org.chromium.chrome" }
 test_manifest_package = "org.chromium.chrome.tests"
 
 chrome_public_jinja_variables = default_chrome_public_jinja_variables +
@@ -260,6 +260,7 @@ android_library("chrome_java") {
     "$google_play_services_package:google_play_services_tasks_java",
     "//base:base_java",
     "//base:jni_java",
+    ":brave_content_java_resources",
     "//chrome/android/third_party/compositor_animator:compositor_animator_java",
     "//chrome/android/webapk/libs/client:client_java",
     "//chrome/android/webapk/libs/common:common_java",
@@ -1018,6 +1019,11 @@ android_resources("chrome_public_apk_resources") {
   ]
 }
 
+android_resources("brave_content_java_resources") {
+  resource_dirs = []
+  deps = ["//brave/build/android:android_brave_strings_grd"]
+}
+
 version_resource_dir = "$target_gen_dir/templates/chrome_version_xml/res"
 verson_resource_file = "$version_resource_dir/values/strings.xml"
 process_version("version_xml") {
